Mathlib/Analysis/Convex/Normed.lean

```diff
@@ -130,10 +133,16 @@ instance (priority := 100) NormedSpace.instLocPathConnectedSpace : LocPathConnec
     (convex_ball x r).isPathConnected <| by simp [r_pos]
 #align normed_space.loc_path_connected NormedSpace.instLocPathConnectedSpace
 
+variable (E) in
+lemma Wbtw.dist_add_dist {x y z : P} (h : Wbtw ℝ x y z) :
+  dist x y + dist y z = dist x z := by
+  obtain ⟨a, ⟨ha₀, ha₁⟩, rfl⟩ := h
+  simp [abs_of_nonneg, ha₀, ha₁, sub_mul]
+
 theorem dist_add_dist_of_mem_segment {x y z : E} (h : y ∈ [x -[ℝ] z]) :
     dist x y + dist y z = dist x z := by
-  simp only [dist_eq_norm, mem_segment_iff_sameRay] at *
-  simpa only [sub_add_sub_cancel', norm_sub_rev] using h.norm_add.symm
+  rw [← affineSegment_eq_segment] at h
+  exact Wbtw.dist_add_dist E h
```

Comment:
@YaelDillies, I found this new home for your lemma; does it looks reasonable?

Reply:
Looks fine yeah. I mean, the whole situation is not great, but the answer is to refactor convexity so that betweenness comes much earlier. This is a very reasonable compromise in the meantime.